app/components/ConfirmTrade.tsx

```diff
@@ -65,27 +78,90 @@ export const ConfirmTrade = ({
     ? formatEtherWithFixedDecimals(tokenAmountOut)
     : '';
 
+  const waitingTxToast = (txHash: string) =>
+    toast({
+      children: (
+        <div className="flex items-center space-x-4">
+          <Spinner className="h-5 w-5 shrink-0 animate-spin" />
+          <div>
+            Wating for transaction success{' '}
+            <a
+              href={`${GNOSIS_SCAN_URL}/tx/${txHash}`}
+              target="_blank"
+              className="inline-block underline"
+            >
+              {shortenAddress(txHash)}
+            </a>
+          </div>
+        </div>
+      ),
+    });
+
+  const succesTxToast = (txHash: string, token: Token | Outcome) => {
+    const symbol = token.symbol || '';
+
+    successToast({
+      children: (
+        <>
+          {symbol} approved successfully{' '}
+          <a
+            href={`${GNOSIS_SCAN_URL}/tx/${txHash}`}
+            target="_blank"
+            className="inline-block underline"
+          >
+            {shortenAddress(txHash)}
+          </a>
+        </>
+      ),
+    });
+  };
+
   const approveToken = async () => {
+    setIsApproving(true);
     const inToken = swapState.inToken as Token;
-    submitTx({
+
+    writeContract(config, {
       abi: erc20Abi,
       address: inToken.address,
       functionName: 'approve',
       args: [marketId, amountWei],
-    }).then(() => {
-      onApprove();
-    });
+    })
+      .then(async txHash => {
+        waitingTxToast(txHash);
+
+        await waitForTransactionReceipt(config, {
+          hash: txHash,
+        });
+
+        succesTxToast(txHash, inToken);
```

Comment:
The tx can fail (not cancelled), right? If it fails, it will be caught by `catch()`  and the user will have to submit a new tx. Probably we should show an error toast.

Reply:
Done, thanks for the feedback!